src/tunnel.c

```diff
@@ -158,6 +168,11 @@ static int pppd_run(struct tunnel *tunnel)
 
 	slave_stderr = dup(STDERR_FILENO);
 
+	if (slave_stderr < 0) {
+		log_error("slave stderr %d\n", slave_stderr);
```

Comment:
I suggest using `strerror(errno)` in the error message.

Reply:
done.